src/modules/personenkontext/domain/personenkontext-workflow.ts

```diff
@@ -0,0 +1,344 @@
+        personenkontexte: DbiamPersonenkontextBodyParams[],
+    ): Promise<Personenkontext<true>[] | PersonenkontexteUpdateError> {
+        const pkUpdate: PersonenkontexteUpdate = this.dbiamPersonenkontextFactory.createNewPersonenkontexteUpdate(
+            personId,
+            lastModified,
+            count,
+            personenkontexte,
+        );
+        const updateResult: Personenkontext<true>[] | PersonenkontexteUpdateError = await pkUpdate.update();
+
+        if (updateResult instanceof PersonenkontexteUpdateError) {
+            return updateResult;
+        }
+        return updateResult;
+    }
+
+    // Checks if the rolle can be assigned to the target organisation
+    public async checkReferences(organisationId: string, rolleId: string): Promise<Option<DomainError>> {
+        const [orga, rolle]: [Option<OrganisationDo<true>>, Option<Rolle<true>>] = await Promise.all([
+            this.organisationRepository.findById(organisationId),
+            this.rolleRepo.findById(rolleId),
+        ]);
+
+        if (!orga) {
+            return new EntityNotFoundError('Organisation', organisationId);
+        }
+
+        if (!rolle) {
+            return new EntityNotFoundError('Rolle', rolleId);
+        }
+        // Can rolle be assigned at target orga
+        const canAssignRolle: boolean = await rolle.canBeAssignedToOrga(organisationId);
+        if (!canAssignRolle) {
+            return new EntityNotFoundError('Rolle', rolleId); // Rolle does not exist for the chosen organisation
+        }
+
+        //The aimed organisation needs to match the type of role to be assigned
+        const organisationMatchesRollenart: OrganisationMatchesRollenart = new OrganisationMatchesRollenart();
+        if (!organisationMatchesRollenart.isSatisfiedBy(orga, rolle)) {
+            return new RolleNurAnPassendeOrganisationError();
+        }
+
+        return undefined;
+    }
+
+    public async checkPermissions(
+        permissions: PersonPermissions,
+        organisationId: string,
+    ): Promise<Option<DomainError>> {
+        // Check if logged in person has permission
+        const hasPermissionAtOrga: boolean = await permissions.hasSystemrechtAtOrganisation(organisationId, [
+            RollenSystemRecht.PERSONEN_VERWALTEN,
+        ]);
+
+        // Missing permission on orga
+        if (!hasPermissionAtOrga) {
+            return new MissingPermissionsError('Unauthorized to manage persons at the organisation');
+        }
+
+        return undefined;
+    }
+
+    public async findSchulstrukturknoten(
+        rolleId: string,
+        sskName: string,
+        limit?: number,
+        excludeKlassen: boolean = false,
+    ): Promise<OrganisationDo<true>[]> {
+        this.selectedRolleId = rolleId;
+
+        let organisationsFoundByName: OrganisationDo<boolean>[] = [];
+        let total: number = 0;
+
+        if (excludeKlassen) {
+            const scope: OrganisationScope = new OrganisationScope();
```

Comment:
Scopes haben eine direkte Abhängigkeit auf MikroORM und sollen daher nur in der Persistence-Schicht benutzt werden.